dev-tools/mage/manifest/manifest.go

```diff
@@ -85,7 +85,7 @@ func resolveManifestPackage(project tools.Project, pkg string, reqPackage string
 func DownloadComponentsFromManifest(manifest string, platforms []string, platformPackages map[string]string, dropPath string) error {
 	componentSpec := map[string][]string{
 		"apm-server":            {"apm-server"},
-		"beats":                 {"auditbeat", "filebeat", "heartbeat", "metricbeat", "osquerybeat", "packetbeat"},
+		"beats":                 {"agentbeat"},
 		"cloud-defend":          {"cloud-defend"},
```

Comment:
(Similar issue as the one above) This bit of code can be used with manifests older than 8.14 (8.13.x for example) and the repackaging can happen on any branch since it's independent from the actual code...

This change can break if a repackage for 8.13 is launched using the magefile from main....
Did you check with the early-agent-releases what this change implies for them ? cc @pierrehilbert @cmacknz

Reply:
> > on which branch are we planning to call the mage downloadManifest target if we need to create an agent early release ?
> 
> Once we're fully ready, the Independent Agent Staging build will run the `elastic-agent-package` pipeline on the "Stack Branch" for the release. So if we are building a `8.13.2+build202404171057` staging build, then it will run on the `8.13` branch for the `elastic-agent-package` pipeline.
> 
> We're going to start doing regular/daily builds starting soon on `8.13`, followed by `8.14` sometime later.

On 8.13 we don't have the new target to use elastic-agent-core artifacts that was introduced with #4403
The assumption I had when implementing that was that the pipeline could be triggered from anywhere (not true anymore after this change) and we already partially backported #4403 (for the env var rename) which will make the backport more complicated. I guess we need to sync on where you expect the `packageUsingDRA` mage target to exist

Reply:
Opened https://github.com/elastic/elastic-agent/pull/4589 to backport the repackaging logic on 8.13 branch